src/main/java/com/avispl/symphony/dal/avdevices/microphone/sennheiser/tcc2/comom/SennheiserPropertiesList.java

```diff
@@ -0,0 +1,110 @@
+	IPV4_ADDRESS(SennheiserConstant.IPV4_ADDRESS, SennheiserCommandList.IPV4_ADDRESS, false),
+	IPV4_DEFAULT_GATEWAY(SennheiserConstant.IPV4_DEFAULT_GATEWAY, SennheiserCommandList.IPV4_DEFAULT_GATEWAY, false),
+	IPV4_NETMASK(SennheiserConstant.IPV4_NETMASK, SennheiserCommandList.IPV4_NETMASK, false),
+	MAC_ADDRESS(SennheiserConstant.MAC_ADDRESS, SennheiserCommandList.MAC_ADDRESS, false),
+	IPV4_INTERFACE_NAME(SennheiserConstant.IPV4_INTERFACE_NAME, SennheiserCommandList.IPV4_INTERFACE_NAME, false),
+	IDENTIFY_DEVICE(SennheiserConstant.IDENTIFY_DEVICE, SennheiserCommandList.IDENTIFY_DEVICE_CONTROLLING, true),
+	LED_BRIGHTNESS(SennheiserConstant.LED_BRIGHTNESS, SennheiserCommandList.LED_BRIGHTNESS_CONTROLLING, true),
+	DEVICE_RESTART(SennheiserConstant.DEVICE_RESTART, SennheiserCommandList.DEVICE_RESTART_CONTROLLING, true),
+	AUDIO_MUTE(SennheiserConstant.AUDIO_MUTE, SennheiserCommandList.AUDIO_MUTE_CONTROLLING, true),
+	VOICE_LIFT(SennheiserConstant.VOICE_LIFT, SennheiserCommandList.VOICE_LIFT_CONTROLLING, true),
+	INPUT_LEVEL_GAIN(SennheiserConstant.INPUT_LEVEL_GAIN, SennheiserCommandList.INPUT_LEVEL_GAIN_CONTROLLING, true),
+	MIC_ON_LED_COLOR(SennheiserConstant.MIC_ON_LED_COLOR, SennheiserCommandList.MIC_ON_LED_COLOR_CONTROLLING, true),
+	MIC_MUTE_LED_COLOR(SennheiserConstant.MIC_MUTE_LED_COLOR, SennheiserCommandList.MIC_MUTE_LED_COLOR_CONTROLLING, true),
+	LED_CUSTOM_COLOR(SennheiserConstant.LED_CUSTOM_COLOR, SennheiserCommandList.LED_CUSTOM_COLOR_CONTROLLING, true),
+	FAR_END_ACTIVITY_LED_MODE(SennheiserConstant.FAR_END_ACTIVITY_LED_MODE, SennheiserCommandList.FAR_END_ACTIVITY_LED_MODE_CONTROLLING, true),
+	;
+	private final String name;
+	private final String command;
+	private boolean isControl;
+
+	/**
+	 * SennheiserPropertiesList instantiation
+	 *
+	 * @param name property name
+	 * @param command property command
+	 * @param isControl controlling or monitoring
+	 */
+	SennheiserPropertiesList(String name, String command, boolean isControl) {
+		this.name = name;
+		this.command = command;
+		this.isControl = isControl;
+	}
+
+	/**
+	 * Retrieves {@link #name}
+	 *
+	 * @return value of {@link #name}
+	 */
+	public String getName() {
+		return name;
+	}
+
+	/**
+	 * Retrieves {@link #command}
+	 *
+	 * @return value of {@link #command}
+	 */
+	public String getCommand() {
+		return command;
+	}
+
+	/**
+	 * Retrieves {@code {@link #isControl}}
+	 *
+	 * @return value of {@link #isControl}
+	 */
+	public boolean isControl() {
+		return isControl;
+	}
+
+	/**
+	 * This method is used to get device metric group by name
+	 *
+	 * @param name is the name of device metric group that want to get
+	 * @return DevicesMetricGroup is the device metric group that want to get
+	 */
+	public static SennheiserPropertiesList getByName(String name) {
+		Optional<SennheiserPropertiesList> property = Arrays.stream(SennheiserPropertiesList.values()).filter(group -> group.getName().equals(name)).findFirst();
```

Comment:
This code can be refactor in concise way:
return Arrays.stream(SennheiserPropertiesList.values()).filter(group -> group.getName().equals(name)).findFirst().orElseThrow( new IllegalStateException(String.format("control group %s is not supported.", name));
Please check in another enum also